### PR TITLE
Merged seo and jsonld modules

### LINKED ISSUE #1536: seo and jsonld modules should be merged
The [seo](https://github.com/w3c/respec/tree/develop/src/core/seo.js) module and the newly added [jsonld](https://github.com/w3c/respec/tree/develop/src/w3c/jsonld.js) should be merged. They share code and serve a very similar purpose.

### PR DESCRIPTION
**Pull request by goelrohan6**
Closes #1536
w3c/respec#1536

### CODE REVIEW

File: src/core/seo.js [hunk 1]
```diff
@@ -14,6 +15,76 @@ export async function run(conf, doc, cb) {
     return; // no abstract, so nothing to do
   }
   insertMetaDescription(firstParagraph);
+  if (conf.doJsonLd) {
```
Comment (marcoscaceres) [inline src/core/seo.js]: If suggest moving all this into it's own `function addJSONLDInfo(conf) { ... }` instead.

Comment (goelrohan6) [reply src/core/seo.js]: done @marcoscaceres, kindly check

**Comment by goelrohan6** [comment]
@marcoscaceres merging w3c/jsonld-spec.js into core/seo-spec.js .

Or is it w3c/seo-spec.js as you said?

**Comment by goelrohan6** [comment]
pushing to the same branch @marcoscaceres

**Comment by goelrohan6** [comment]
your one commit 'cleanup' would get lost

**Comment by marcoscaceres** [comment]
`git pull` doesn’t pull it down? I’d assumed it would.

**Comment by OrionStar25** [comment]
Git pull works!

**Comment by goelrohan6** [comment]
@marcoscaceres I've made the final changes

need your review after that we are good to go

**Comment by marcoscaceres** [comment]
Ah, did you squish the history? Seems my cleanup got trashed?

**Comment by goelrohan6** [comment]
I'm so sorry, I did.

You can tell me I can make the changes.

**Comment by marcoscaceres** [comment]
I guess it doesn't matter. They were mostly cosmetic changes. Can redo them another time.

**Comment by marcoscaceres** [comment]
One small thing, can you remove the json-ld.js test from:
https://github.com/w3c/respec/blob/develop/tests/testFiles.json

Thanks!

**Comment by marcoscaceres** [comment]
Need to wait till TravisCI gives it an ok ;)

**Comment by marcoscaceres** [comment]
@goelrohan6, can you `git pull` my latest changes and then give this a rebase with develop? Should hopefully be all good then.

**Comment by goelrohan6** [comment]
@marcoscaceres doing it

**Comment by goelrohan6** [comment]
@marcoscaceres I've rebased develop into this branch, should I push?

**Comment by marcoscaceres** [comment]
yep, give it a push.

**Comment by goelrohan6** [comment]
@marcoscaceres can you check

**Comment by marcoscaceres** [comment]
Seems nothing was pushed... try: `git push -f`

**Comment by goelrohan6** [comment]
@marcoscaceres I rebased develop into this branch and pushed develop.

Was it supposed  to be done other way round, I'm confused with rebase

**Comment by marcoscaceres** [comment]
Ah, push this branch instead.

**Comment by goelrohan6** [comment]
Can you check now?

**Comment by marcoscaceres** [comment]
still nothing :) It's ok... don't worry, the tests are passing... I'll double that everything is fine before releasing.

**Pull request merged**